pkg/apis/batch/v2alpha1/types.go

```diff
@@ -250,6 +250,16 @@ type CronJobSpec struct {
 	// JobTemplate is the object that describes the job that will be created when
 	// executing a CronJob.
 	JobTemplate JobTemplateSpec `json:"jobTemplate" protobuf:"bytes,5,opt,name=jobTemplate"`
+
+	// The number of successful finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"varint,6,opt,name=successfulJobsHistoryLimit"`
```

Comment:
@smarterclayton @soltysh is adding the defaults only for Deployments created in new clusters. Older Deployments shouldn't be pruned. https://github.com/kubernetes/kubernetes/pull/39683